sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/ResourceBuilderTest.java

```diff
@@ -85,6 +86,51 @@ public void testNewFromResourceSpecRefined() {
         }
     }
 
+    /*
+        name: "cpus"
+        type: SCALAR
+        scalar {
+          value: 1.0
+        }
+        reservations {
+          role: "base-role"
+          type: STATIC
+        }
+        reservations {
+          principal: "test-principal"
+          labels {
+            labels {
+              key: "resource_id"
+              value: "a395f14b-3cc8-4009-9dc4-51838b423aed"
+            }
+          }
+          role: "test-role"
+          type: DYNAMIC
+        }
+    */
+    @Test
+    public void testRefineStaticResource() {
+        ResourceRefinementCapabilityContext context = new ResourceRefinementCapabilityContext(Capabilities.getInstance());
+        try {
+            ResourceSpec resourceSpec = new DefaultResourceSpec(
+                    "cpus",
+                    value,
+                    TestConstants.ROLE,
+                    TestConstants.PRE_RESERVED_ROLE,
+                    TestConstants.PRINCIPAL,
+                    "CPUS_ENV_KEY");
+            ResourceBuilder resourceBuilder = ResourceBuilder.fromSpec(resourceSpec, Optional.empty());
+
+            Protos.Resource resource = resourceBuilder.build();
+            Assert.assertEquals(2, resource.getReservationsCount());
+            validateScalarResourceRefined(TestConstants.PRE_RESERVED_ROLE, resource);
```

Comment:
I don't think helper functions like this are valuable in tests. For the most part I want to be able to read a single test case and know what conditions are under test, `validateScalarResourceRefined` doesn't do that and I have to jump around to understand it. If un-DRYing it is onerous I think it's a sign that further decomposition is warranted.